ts/src/base/Exchange.ts

```diff
@@ -1542,6 +1542,35 @@ export default class Exchange {
         };
     }
 
+    currencyStructure () {
+        return {
+            'info': undefined,
+            'id': undefined,
+            'numericId': undefined,
+            'code': undefined,
+            'precision': undefined,
+            'type': undefined,
+            'margin': undefined,
```

Comment:
@samgermain @ttodua question, why `margin` and `type` here?

Reply:
iirc type is used to have `fiat` or `crypto` string values (or undefined).
margin might be if the coin is enabled for margin borrow (idk if this should be in currencies, as we dont have `fetchBorrowableCurrencies` or alike, so maybe that's why it was used here. I am not aware of the context which exchanges set that field, but might (or not) be good if that was supported there

Reply:
@ttodua so I'm ok with `type` but yeah I think `margin` does not fit here

Reply:
I don't have a complete view about `margin`, as some exchanges (like gate iirc) does not make currency borrowable to specific market (i.e. say having isolated margin for `XYZ/USDT` specifically, but not i.e. `XYZ/GUSD`, instead it has a page where you borrow `XYZ` coin and then you can spend it on any marginable pair for XYZ.
So, at one hand the coin might have `margin` field, but on other hand, i rarely seen exchanges providing that value in currencies.

So, maybe we can wait to collect the common responses across exchanges to add (or not) that field eventually

Reply:
ok, I've removed `margin`. whenever you can, it is ready for merge.
@samgermain if there is specific exchange that uses `margin`, lets add that inside implementation of that specific exchange. 
in future we might overlook this.